src/components/shared/ReactFlow/FlowCanvas/ConfirmationDialogs/BulkUpdateConfirmationDialog.tsx

```diff
@@ -0,0 +1,31 @@
+import type { Node } from "@xyflow/react";
+
+import type { InputSpec } from "@/utils/componentSpec";
+
+import { thisCannotBeUndone } from "./shared";
+
+export function getBulkUpdateConfirmationDetails(
+  nodesToUpdate: Node[],
+  allLostInputs: InputSpec[],
+) {
+  const hasLostInputs = allLostInputs.length > 0;
+
+  const title = "Upgrade Components";
+  const description = "";
+  const content = (
+    <div className="text-sm">
+      <p>Are you sure you want to update these nodes?</p>
+      <p>{`This will upgrade ${nodesToUpdate.length} nodes to new versions from their source URL.`}</p>
```

Comment:
```suggestion
      <p>This will upgrade {nodesToUpdate.length} nodes to new versions from their source URL.</p>
```

I know we discussed it yesterday but I think removing the template literal will make it easier to read. The last one suffered from the escaped characters, but this does not

Reply:
This one is fine to read. I'll update